warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java

```diff
@@ -229,171 +236,306 @@
+        this.setUniqueFields(null == other.getUniqueFields() ? null : Sets.newHashSet(other.getUniqueFields()));
+        this.setCacheModel(other.getCacheModel());
+        this.setTrackSizes(other.isTrackSizes());
+        this.setContentFieldNames(null == other.getContentFieldNames() ? null : Lists.newArrayList(other.getContentFieldNames()));
+    }
     
     /**
-     * Used to determine how many ranges the ThreadedRangeBundler should buffer before returning a range to the caller
+     * Delegates deep copy work to appropriate constructor, sets additional values specific to the provided ShardQueryLogic
+     *
+     * @param logic
+     *            - a ShardQueryLogic instance or subclass
      */
-    protected int numRangesToBuffer = 0;
+    public ShardQueryConfiguration(ShardQueryLogic logic) {
+        this(logic.getConfig());
+        
+        // Setters that would have been picked up in a super(logic) call
+        this.setTableName(logic.getTableName());
+        this.setMaxQueryResults(logic.getMaxResults());
+        this.setMaxRowsToScan(logic.getMaxRowsToScan());
+        this.setUndisplayedVisibilities(logic.getUndisplayedVisibilities());
+        this.setBaseIteratorPriority(logic.getBaseIteratorPriority());
+    }
     
     /**
-     * Used to determine how long to allow the ThreadedRangeBundler to buffer ranges before returning a range to the caller
+     * Factory method that instantiates an fresh ShardQueryConfiguration
+     *
+     * @return - a clean ShardQueryConfiguration
      */
-    protected long rangeBufferTimeoutMillis = 0;
+    public static ShardQueryConfiguration create() {
+        return new ShardQueryConfiguration();
+    }
     
     /**
-     * Used to determine the poll interval when buffering ranges in ThreadedRangeBundler
+     * Factory method that returns a deep copy of the provided ShardQueryConfiguration
+     *
+     * @param other
+     *            - another instance of a ShardQueryConfiguration
+     * @return - copy of provided ShardQueryConfiguration
      */
-    protected long rangeBufferPollMillis = 100;
+    public static ShardQueryConfiguration create(ShardQueryConfiguration other) {
+        return new ShardQueryConfiguration(other);
+    }
     
     /**
-     * Used to determine the maximum number of query ranges to generate per tier when performing a geowave query.
+     * Factory method that creates a ShardQueryConfiguration deep copy from a ShardQueryLogic
+     *
+     * @param shardQueryLogic
+     *            - a configured ShardQueryLogic
+     * @return - a ShardQueryConfiguration
      */
-    protected int geoWaveMaxExpansion = 800;
+    public static ShardQueryConfiguration create(ShardQueryLogic shardQueryLogic) {
+        
+        ShardQueryConfiguration config = create(shardQueryLogic.getConfig());
+        
+        if (-1 == shardQueryLogic.getMaxResults()) {
```

Comment:
-1 is magic numbery, is there a constant somewhere or maybe maxResults < 0?

Reply:
Good catch, 'maxResults < 0' is cleaner.